File: src/main/java/kajipoi/entity/Member.java
package kajipoi.entity;

import kajipoi.domain.Name;
import lombok.Value;
import lombok.experimental.Accessors;
import org.seasar.doma.Entity;
import org.seasar.doma.GeneratedValue;
import org.seasar.doma.GenerationType;
import org.seasar.doma.Id;
import org.seasar.doma.jdbc.entity.NamingType;

/**
 * メンバーリソースを表すエンティティ。
 *
 * @author syobochim
 * @since 1.0
 */
@Entity(naming = NamingType.LOWER_CASE, immutable = true)
@Value(staticConstructor = "of")
@Accessors(fluent = true)
public class Member {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private final Id memberId;

    private final Name memberName;
}

File: src/main/kotlin/kajipoi/entity/Member.kt
package kajipoi.entity

import kajipoi.domain.Id
import kajipoi.domain.Name
import org.seasar.doma.Entity
import org.seasar.doma.GeneratedValue
import org.seasar.doma.GenerationType
import org.seasar.doma.jdbc.entity.NamingType

/**
 * @author syobochim
 */
@Entity(naming = NamingType.LOWER_CASE, immutable = true)
class Member(@GeneratedValue(strategy = GenerationType.IDENTITY) val memberId: Id,
             val memberName: Name)